### ISSUE TITLE
Indices replicated with CCR are randomly paused when ILM are enabled

### ISSUE THREAD

**Issue by medcl** [elastic/elasticsearch#63681]
**Elasticsearch version** (`bin/elasticsearch --version`):
Tested with v7.9.2

**Plugins installed**: []
None

**Steps to reproduce**:

Two cluster:
- Cluster dev: 192.168.3.98
- Cluster local: localhost

Same version, v7.9.2; CCR was configured to replicated index from `192.168.3.98` to `localhost`.

1.  on dev cluster(192.168.3.98)

```
PUT _cluster/settings
{
  "transient": {
    "indices.lifecycle.poll_interval":"1m"
  }}
  
PUT _ilm/policy/medcl_ilm_test
{
  "policy": {
    "phases": {
      "hot": {
        "min_age": "0ms",
        "actions": {
          "rollover": {
            "max_age": "1m",
            "max_size": "10mb",
            "max_docs": 1000
          },
          "set_priority": {
            "priority": 100
          },
          "unfollow": {}
        }
      },
      "delete": {
        "min_age": "5m",
        "actions": {
          "delete": {
            "delete_searchable_snapshot": true
          }
        }
      }
    }
  }
}


PUT _template/medcl_template_test
{
  "index_patterns": ["medcl-index-test-*"],
  "settings": {
    "number_of_shards": 1,
    "refresh_interval": "1s", 
    "number_of_replicas": 0,
    "index.lifecycle.name": "medcl_ilm_test", 
    "index.lifecycle.rollover_alias": "medcl-index" 
  }
}

PUT medcl-index-test-000001
{
  "aliases": {
    "medcl-index":{
      "is_write_index": true 
    }
  }
}
```

2.  on local cluster(localhost)

```
PUT _cluster/settings
{
  "persistent": {
    "cluster": {
      "remote": {
        "dev": {
          "skip_unavailable": true,
          "mode": "sniff",
          "proxy_address": null,
          "proxy_socket_connections": null,
          "server_name": null,
          "seeds": [
            "192.168.3.98:9300"
          ],
          "node_connections": 3
        }
      }
    }
  }
}

PUT _ilm/policy/medcl_ilm_test
{
  "policy": {
    "phases": {
      "hot": {
        "actions": {
          "unfollow": {}
        }
      }
      ,
      "delete": {
        "min_age": "5m",
        "actions": {
          "delete": {
            "delete_searchable_snapshot": true
          }
        }
      }
    }
  }
}

PUT _cluster/settings
{
  "transient": {
    "indices.lifecycle.poll_interval":"1m"
  }
}

PUT /_ccr/auto_follow/medcl_test_auto_follow_pattern
{
  "remote_cluster" : "dev",
  "leader_index_patterns" :
  [
    "medcl-index-test-*"
  ],
  "follow_index_pattern" : "{{leader_index}}-follower"
}
```

3.  start sending testing data to `medcl-index`, after running for a while, the problem show up

4.  if you check the follower indices in `localhost` cluster, you will see many indices with `0` docs

```
GET _cat/indices/medcl*?v&s=index
```
![image](https://user-images.githubusercontent.com/64487/96004031-7a3d6480-0e6d-11eb-981c-e7ee3d9f16b9.png)
![image](https://user-images.githubusercontent.com/64487/96004809-680ff600-0e6e-11eb-8d98-72c454803458.png)

As you can see, there are many follower indices are not well replicated. actually they are not started the replication, i seems executed `unfollow` before finished the CCR replication step.

And if you check the follower index's ilm status, 
```
GET medcl-index-test-000010-follower/_ilm/explain
{
  "indices" : {
    "medcl-index-test-000010-follower" : {
      "index" : "medcl-index-test-000010-follower",
      "managed" : true,
      "policy" : "medcl_ilm_test",
      "lifecycle_date_millis" : 1602664972436,
      "age" : "5.89h",
      "phase" : "hot",
      "phase_time_millis" : 1602664982032,
      "action" : "unfollow",
      "action_time_millis" : 1602664982032,
      "step" : "wait-for-indexing-complete",
      "step_time_millis" : 1602664982032,
      "step_info" : {
        "message" : "waiting for the [index.lifecycle.indexing_complete] setting to be set to true on the leader index, it is currently [false]",
        "index.lifecycle.indexing_complete" : false
      },
      "phase_execution" : {
        "policy" : "medcl_ilm_test",
        "phase_definition" : {
          "min_age" : "0ms",
          "actions" : {
            "unfollow" : { }
          }
        },
        "version" : 2,
        "modified_date_in_millis" : 1602662527140
      }
    }
  }
}
```

And check for leader's index:
```
GET medcl-index-test-000010/_ilm/explain
{
  "indices" : {
    "medcl-index-test-000010" : {
      "index" : "medcl-index-test-000010",
      "managed" : true,
      "policy" : "medcl_ilm_test",
      "lifecycle_date_millis" : 1602664976942,
      "age" : "5.9h",
      "phase" : "delete",
      "phase_time_millis" : 1602665276948,
      "action" : "delete",
      "action_time_millis" : 1602665276948,
      "step" : "wait-for-shard-history-leases",
      "step_time_millis" : 1602665276948,
      "step_info" : {
        "message" : "this index is a leader index; waiting for all following indices to cease following before proceeding"
      },
      "phase_execution" : {
        "policy" : "medcl_ilm_test",
        "phase_definition" : {
          "min_age" : "5m",
          "actions" : {
            "delete" : {
              "delete_searchable_snapshot" : true
            }
          }
        },
        "version" : 10,
        "modified_date_in_millis" : 1602645128372
      }
    }
  }
}
```
And it says It is waiting for following index to cease but actually the following is not started.

the follower index is not able to resume via API:
```
POST medcl-index-test-000139-follower/_ccr/resume_follow
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "the leader index settings [{\"index.lifecycle.indexing_complete\":\"true\",\"index.lifecycle.name\":\"medcl_ilm_test\",\"index.lifecycle.rollover_alias\":\"medcl-index\",\"index.number_of_shards\":\"1\"}] and follower index settings [{\"index.lifecycle.name\":\"medcl_ilm_test\",\"index.lifecycle.rollover_alias\":\"medcl-index\",\"index.number_of_shards\":\"1\"}] must be identical"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "the leader index settings [{\"index.lifecycle.indexing_complete\":\"true\",\"index.lifecycle.name\":\"medcl_ilm_test\",\"index.lifecycle.rollover_alias\":\"medcl-index\",\"index.number_of_shards\":\"1\"}] and follower index settings [{\"index.lifecycle.name\":\"medcl_ilm_test\",\"index.lifecycle.rollover_alias\":\"medcl-index\",\"index.number_of_shards\":\"1\"}] must be identical"
  },
  "status" : 400
}

```

The index withe these issue are randomly picked.

**Comment by andreidan** [comment]
@medcl thanks for reporting this problem and for providing the setup details. Under normal operations (basically executing the steps to reproduce) I was not able to reproduce the behaviour you're seeing for quite some time (bulk indexing sporadically executed against the leader): 
<img width="1140" alt="ccr_kibana_550" src="https://user-images.githubusercontent.com/1217601/102088662-23bfb880-3e13-11eb-8e89-65cbc2318ee4.png">

However, if I paused the following _before_ the leader was marked as `indexing_complete` and then resumed the following I was able to reproduce the exception you had when trying to execute `_ccr/resume_follow`. Namely:
```
        "reason": "the leader index settings [{\"index.lifecycle.indexing_complete\":\"true\",\"index.lifecycle.name\":\"medcl_ilm_test\",\"index.lifecycle.rollover_alias\":\"medcl-index\",\"index.number_of_shards\":\"1\"}] and follower index settings [{\"index.lifecycle.name\":\"medcl_ilm_test\",\"index.lifecycle.rollover_alias\":\"medcl-index\",\"index.number_of_shards\":\"1\"}] must be identical"
```
In order to resume the following, the leader having already had completed indexing (ie. `index.lifecycle.indexing_complete : true`) I manually configured `index.lifecycle.indexing_complete: true` on the follower and then resumed follow and that was successful. 

Is the `pause_follow` / `resume_follow` scenario something that might've happened in your environment?
